scipy/special/cephes/bdtr.c

```diff
@@ -148,103 +148,89 @@
 
 #include "mconf.h"
 
-double bdtrc(k, n, p)
-int k, n;
-double p;
+double bdtrc(double k, double n, double p)
 {
     double dk, dn;
 
     if (npy_isnan(p)) {
 	return NPY_NAN;
     }
-    if ((p < 0.0) || (p > 1.0)) {
-	goto domerr;
+    if (p < 0.0 || p > 1.0 || n < k) {
+	sf_error("bdtrc", SF_ERROR_DOMAIN, NULL);
+	return NPY_NAN;
     }
     if (k < 0) {
 	return 1.0;
     }
-
-    if (n < k) {
-      domerr:
-	sf_error("bdtrc", SF_ERROR_DOMAIN, NULL);
-	return NPY_NAN;
-    }
-
-    if (k == n)
```

Comment:
I assume this path was dropped because `incbet` is doing to handle it correctly below anyway?

Reply:
I don't think `incbet` was handling correctly. See comment above about `2,2,0.5` going to a `nan`